### PR TITLE
[gl] extract JSContextRef from jsi runtime

### LINKED ISSUE #4051: [sdk33][ios] Patch EXGL until we convert it to a Turbomodule
EXGL expects a JSGlobalContextRef, which RN's CxxBridge no longer exposes. We need to get this from the C++ layer. In the future EXGL should be a Turbomodule that has direct JS-native access.

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why
Fixes #4051 

# How

Extract from jsi::Runtime using constant offset

# Test Plan

### CODE REVIEW

File: packages/@unimodules/react-native-adapter/ios/UMReactNativeAdapter/Services/UMReactNativeAdapter.m [hunk 1]
```diff
@@ -130,7 +132,11 @@ - (void)unregisterAppLifecycleListener:(id<UMAppLifecycleListener>)listener
 
 - (JSGlobalContextRef)javaScriptContextRef
 {
-  return _bridge.jsContextRef;
+  if ([_bridge respondsToSelector:@selector(jsContextRef)]) {
+    return _bridge.jsContextRef;
+  } else {
+    return *(((JSGlobalContextRef*)(_bridge.runtime )) + 1);
```
Comment (sjchmiela) [inline packages/@unimodules/react-native-adapter/ios/UMReactNativeAdapter/Services/UMReactNativeAdapter.m]: Would you, please, add some explanatory comment here why do we have to do such magic?

File: packages/@unimodules/react-native-adapter/ios/UMReactNativeAdapter/Services/UMReactNativeAdapter.m [hunk 2]
```diff
@@ -19,6 +20,7 @@ @interface UMReactNativeAdapter ()
 @interface RCTBridge ()
 
 - (JSGlobalContextRef)jsContextRef;
+- (void*) runtime;
```
Comment (sjchmiela) [inline packages/@unimodules/react-native-adapter/ios/UMReactNativeAdapter/Services/UMReactNativeAdapter.m]:
```suggestion
- (void *)runtime;
```

**Review by sjchmiela — COMMENTED** [review]
😲

File: packages/@unimodules/react-native-adapter/ios/UMReactNativeAdapter/Services/UMReactNativeAdapter.m (hunk 1 above)
Comment (tsapeta) [inline packages/@unimodules/react-native-adapter/ios/UMReactNativeAdapter/Services/UMReactNativeAdapter.m]:
```suggestion
    return *(((JSGlobalContextRef *)(_bridge.runtime)) + 1);
```

**Review by tsapeta — CHANGES_REQUESTED** [review]

**Review by sjchmiela — APPROVED** [review]
Thanks! 👏

**Pull request merged**